flink-ai-flow/ai_flow/plugin_interface/scheduler_interface.py

```diff
@@ -0,0 +1,413 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from abc import ABC, abstractmethod
+from typing import Dict, Text, List, Optional
+from ai_flow.common.configuration import AIFlowConfiguration
+from ai_flow.util import json_utils
+from ai_flow.workflow.workflow import Workflow
+from ai_flow.project.project_description import ProjectDesc
+
+
+class WorkflowInfo(json_utils.Jsonable):
```

Comment:
Can we get rid of the setters in *Info class? I think those fields should be immutable.